docs/3-implementing.md

```diff
@@ -66,3 +73,55 @@ plain field (`foo`) will be of type `keyword`, with no nested field.
 Despite the fact that IDs are often integers in various systems, this is not
 always the case. Since we want to make it possible to map as many data sources
 to ECS as possible, we default to using the `keyword` type for IDs.
+
+## Using non-ECS fields in your event stream
+
+The goal of ECS is to define a common set of fields to make working across various
+streams easier. Each stream will have their own particularities, however, and ECS
+cannot account for all possibilities. It's therefore important that implementers
+feel at ease adding fields and top level objects in their schema that are
+not defined in ECS. This is expected.
+
+Doing so will however incur a risk of conflicting with future versions of ECS.
+Let's review some strategies to reduce the risk of conflicts happening.
+
+### Conflicts
+
+Before going into strategies, let's define what constitutes a conflict.
+
+#### Conflicts with ECS
+
+* A custom field has the same name as a new ECS field, but an incompatible type:
+  * `float` vs `long`
+  * `text` vs `keyword`
+  * `keyword` vs numeric (`integer`, `long`, etc.)
+  * discrete field vs an object (e.g. a `keyword` field vs a field that contains other fields)
+* A new ECS field has a completely different purpose than the custom field.
+
+The following does **not** constitute a conflict:
+
+* A new field is added to ECS, and matches the name and type of an existing
+  custom field.
+  * E.g. you define `process.xpid` as a `keyword` field, and ECS adds
+    `process.xpid` as a `keyword` field.
+* A custom `keyword` field gets defined with the same purpose, with a different
+  `ignore_above` value than a new ECS field.
+
+#### Conflicts with Third Parties
+
+Elastic is developing ECS and is adjusting their solutions to leverage ECS.
```

Comment:
I think we put it a bit too broad here. We should care about not conflicting with ECS, third parties will be a problem of it's own but not worth mention here as otherwise it gets too complex.